File: FinalProject.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.stream.LongStream;

// Not completed CGPA value yet to be printed out 
//calculation for GPA in both semesters completed

public class Main {

ArrayList<String> Course_name_arr = new ArrayList<String>();
ArrayList<Integer> Course_unit_arr = new ArrayList<Integer>();
ArrayList<Double> student_assessment_score_arr = new ArrayList<Double>();
ArrayList<Double> student_exam_score_arr = new ArrayList<Double>();
ArrayList<Double> add_exam_and_assessment_arr = new ArrayList<Double>();
ArrayList<String> grade_arr = new ArrayList<String>();
ArrayList<Integer> point_arr = new ArrayList<Integer>();
ArrayList<Integer> product_unit_points_arr = new ArrayList<Integer>();
ArrayList<Integer> GPA_arr = new ArrayList<Integer>();

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        System.out.println("Enter your Matric number ::");               //collects matric number
        String matric_number = scan.nextLine();
        System.out.println("Enter your Full_Name ::");                      //collects Student name
        String student_name = scan.nextLine();
        //Runs the First semester method
        FirstSemester();
        // Runs the Second semester method
        SecondSemester();
        System.out.println(student_name + " with Matric Number ::" + matric_number);
        //Runs the First GPA method
       
       

    }

   

    static void FirstSemester() {
        System.out.println("'\n\n\t\t\t-----------+---This is for your First Semester Courses--+------- \n\n");
        InformationCollector();
         
        System.out.println("------First Semester Courses has been registered successfully--+------- \n\n");
    }


    static void SecondSemester() {
        System.out.println("\n\n\t\t\t-----------+---This is for your Second Semester Courses--+------- \n\n");
        InformationCollector();
   
        System.out.println("------Second Semester Courses has been registered successfully--+------- \n\n");
    }




                static void InformationCollector() {

                    Scanner scan = new Scanner(System.in);
                    System.out.println("Enter the number of courses you are offering (e.g. 7) ::");
                    int Total_Num_of_Courses = scan.nextInt();

                    //CLass and object created to access values
                        Main myObj = new Main();

                        //for loop
                    for(int i=0;i<Total_Num_of_Courses;i++) {
                        Scanner sc = new Scanner(System.in);
                        System.out.println("Enter course name ::");
                        String Course_name = sc.nextLine();
                        System.out.println("Enter the course unit for " + Course_name.toUpperCase() + " ::");
                        int Course_unit = sc.nextInt();
                        System.out.println("Enter the assessment_score for " + Course_name.toUpperCase() + " ::");
                        double student_assessment_score = sc.nextInt();
                        System.out.println("Enter the Examination_score for " + Course_name.toUpperCase() + " ::");
                        double student_exam_score = sc.nextInt();
                        System.out.println("\t\t\t\t\t" + Course_name.toUpperCase() + " has been registered successfully \n\n Let's move to the next Course \n");

            //This sends all infomation collected to the arrays declared global            
                        myObj.Course_name_arr.add(Course_name);
                        myObj.Course_unit_arr.add(Course_unit);
                        myObj.student_assessment_score_arr.add(student_assessment_score);
                        myObj.student_exam_score_arr.add(student_exam_score);
                        double Total_score = student_assessment_score + student_exam_score;
                        myObj.add_exam_and_assessment_arr.add(Total_score);

            //Comapares results in order to know the appropraite grade that should be awarded the student
                        if(Total_score >= 70 && Total_score <= 100) {
                            myObj.grade_arr.add("A");
                            myObj.point_arr.add(5);
                            myObj.product_unit_points_arr.add(Course_unit * 5);
                        }else if(Total_score >= 60 && Total_score <= 69) {
                            myObj.grade_arr.add("B");
                            myObj.point_arr.add(4);
                            myObj.product_unit_points_arr.add(Course_unit * 4);
                        }else if(Total_score >= 50 && Total_score <= 59) {
                            myObj.grade_arr.add("C");
                            myObj.point_arr.add(3);
                            myObj.product_unit_points_arr.add(Course_unit * 3);
                        }else if(Total_score >= 45 && Total_score <= 49) {
                            myObj.grade_arr.add("D");
                            myObj.point_arr.add(2);
                            myObj.product_unit_points_arr.add(Course_unit * 2);
                        }else if(Total_score >= 40 && Total_score <= 44) {
                            myObj.grade_arr.add("E");
                            myObj.point_arr.add(1);
                            myObj.product_unit_points_arr.add(Course_unit * 1);
                        }else if(Total_score >= 0 && Total_score <= 39) {
                            myObj.grade_arr.add("F");
                            myObj.point_arr.add(0);
                            myObj.product_unit_points_arr.add(Course_unit * 0);
                          }
                
                    
                        }


                        //Totals Method was eliminated 
                        // This finds the totals od our array
                        //sum all the course unit
                        int sum = 0;
                        for(int i = 0; i < myObj.Course_unit_arr.size(); i++)
                            sum += myObj.Course_unit_arr.get(i);

                        int Total_course_unit = sum;

                        //sums the course_unit*points
                        int sum1 = 0;
                        for(int i = 0; i < myObj.product_unit_points_arr.size(); i++)
                            sum1 += myObj.product_unit_points_arr.get(i);

                        int Total_unit_points = sum1;

                        //Divides the total points and total course_unit
                    int GPA = Total_unit_points / Total_course_unit ;
                        //sends the divition to the GPA array
                    myObj.GPA_arr.add(GPA);
                    System.out.println("Your GPA for this semester is =  " + GPA);




                        }
}

File: FinalProject.py
#! /usr/bin/python3

#Project made for a student in AHMAN UNIVERSITY,PATIGI KWARA STATE

class Project:
#These are global varaibles which are declared to store data. They are serving as a one time database
    Course_name_arr = []        #Course name array
    Course_unit_arr = []       #Course unit array
    student_assessment_score_arr = []  #Student assessment score array
    student_exam_score_arr = []         #student examination score array
    add_exam_and_assessment_arr = []       #summation of student assessment score and examination score array
    grade_arr = []                          #Grade array
    point_arr = []                      #points array
    product_unit_points_arr = []   #multiplication of course_unit and points array
    GPA_arr = []                    #GPA array

    def __init__(self):
        matric_number = input("Enter your Matric number ::")          #collects matric number
        student_name = input("Enter your Full_Name ::")                   #collects student name
        #Runs the First semester method
        Project.FirstSemester()
        #Runs the Second semester method
        Project.SecondSemester()
        print(student_name + " with Matric Number ::" + str(matric_number))
        #Runs the First GPA method
        Project.GPA()
       
    def GPA():
        #finds the sum of GPA
        Total_GPA = sum(Project.GPA_arr)
        #calculates the CGPA
        CGPA = Total_GPA / 2
        print("----------------\n\nTherefore your CGPA for the 2 semester is = " + str(CGPA))

    def FirstSemester():
        print('\n\n\t\t\t-----------+---This is for your First Semester Courses--+------- \n\n')
        Project.InformationCollector()
        Project.Totals()
        print('------First Semester Courses has been registered successfully--+------- \n\n')
    

    def SecondSemester():
        print('\t\t\t-----------+---This is for your Second Semester Courses--+------- \n\n')
        Project.InformationCollector()
        Project.Totals()
        print('------Second Semester Courses has been registered successfully--+------- \n\n')

    

    def Totals():
        #sum all the course unit
        Total_course_unit = sum(Project.Course_unit_arr)
        #sums the course_unit*points
        Total_unit_points = sum(Project.product_unit_points_arr)
        #Divides the total points and total course_unit
        GPA = Total_unit_points / Total_course_unit 
        #sends the divition to the GPA array
        Project.GPA_arr.append(GPA)
        print("Your GPA for this semester is =  " + str(GPA))
         
           

    def InformationCollector():

        #This collects infomation from the user
        Total_Num_of_Courses = int(input("Enter the number of courses you are offering (e.g. 7) ::"))
        i=0
        while( i < Total_Num_of_Courses):
            Course_name = input("Enter course name ::")
            Course_unit = int(input("Enter the course unit for " + Course_name.upper() + " ::"))
            
            student_assessment_score = int(input("Enter the assessment_score for " + Course_name.upper() + " ::"))
            
            student_exam_score = int(input("Enter the Examination_score for " + Course_name.upper() + " ::"))
            
            print("\t\t\t\t\t" + Course_name.upper() + " has been registered successfully \n\n Let's move to the next Course \n")
            
            #This sends all infomation collected to the arrays declared global
            Project.Course_name_arr.append(Course_name)
            Project.Course_unit_arr.append(Course_unit)
            Project.student_assessment_score_arr.append(student_assessment_score)
            Project.student_exam_score_arr.append(student_exam_score)
            Total_score = student_assessment_score + student_exam_score
            Project.add_exam_and_assessment_arr.append(Total_score)

            #Comapares results in order to know the appropraite grade that should be awarded the student
            if (Total_score >= 70 and Total_score <= 100):
                Project.grade_arr.append("A")
                Project.point_arr.append(5)
                Project.product_unit_points_arr.append(Course_unit * 5)
            elif (Total_score >= 60 and Total_score <= 69): 
                Project.grade_arr.append("B")
                Project.point_arr.append(4)
                Project.product_unit_points_arr.append(Course_unit * 4)
            elif (Total_score >= 50 and Total_score <= 59):
                Project.grade_arr.append("C")
                Project.point_arr.append(3)
                Project.product_unit_points_arr.append(Course_unit * 3)    
            elif (Total_score >= 45 and Total_score <= 49):
                Project.grade_arr.append("D")
                Project.point_arr.append(2)
                Project.product_unit_points_arr.append(Course_unit * 2)  
            elif (Total_score >= 40 and Total_score <= 44):
                Project.grade_arr.append("E")
                Project.point_arr.append(1)
                Project.product_unit_points_arr.append(Course_unit * 1)   
            elif (Total_score >= 0 and Total_score <= 39):
                Project.grade_arr.append("F")
                Project.point_arr.append(0)
                Project.product_unit_points_arr.append(Course_unit * 0)     
            
            i +=1
        



Project()